src/yb/util/oid_generator.h

```diff
@@ -38,4 +38,70 @@ namespace yb {
 
 std::string GenerateObjectId(bool binary_id = false);
 
+class FastRandomNumberGenerator
+{
+public:
+  using result_type = uint64_t;
+
+  FastRandomNumberGenerator() noexcept = default;
+
+  template <class SeedSequence>
+  FastRandomNumberGenerator(SeedSequence &seed_sequence) noexcept
+  {
+    seed(seed_sequence);
+  }
+
+  uint64_t operator()() noexcept
+  {
+    // Uses the xorshift128p random number generation algorithm described in
+    // https://en.wikipedia.org/wiki/Xorshift
+    auto &state_a = state_[0];
+    auto &state_b = state_[1];
+    auto t        = state_a;
+    auto s        = state_b;
+    state_a       = s;
+    t ^= t << 23;        // a
+    t ^= t >> 17;        // b
+    t ^= s ^ (s >> 26);  // c
+    state_b = t;
+    return t + s;
+  }
+
+  // RandomNumberGenerator concept functions required from standard library.
+  // See http://www.cplusplus.com/reference/random/mt19937/
+  template <class SeedSequence>
+  void seed(SeedSequence &seed_sequence) noexcept
+  {
+    seed_sequence.generate(reinterpret_cast<uint32_t *>(state_.data()),
+                           reinterpret_cast<uint32_t *>(state_.data() + state_.size()));
+  }
+
+  static constexpr uint64_t min() noexcept { return 0; }
+
+  static constexpr uint64_t max() noexcept { return std::numeric_limits<uint64_t>::max(); }
+
+private:
+  std::array<uint64_t, 2> state_{};
+};
+
+class OtelRandom
```

Comment:
I suggest you remove otel from the name. In a comment link to the source code where you picked it up from.

Reply:
There is already a Random class, that's why I changed the name

Reply:
Can you call it AUHRandom?

Reply:
yeah